File: data/python/tests/test_autopep8_format_code_with_custom_style.py
import unittest
from function_autopep8_format_code_with_custom_style import CodeFormatter

class TestAutoPep8CustomStyle(unittest.TestCase):

    def test_format_code_basic(self):
        formatter = CodeFormatter()
        code = "def bad_function(x,y):print(x+y)"
        expected_code = "def bad_function(x,y):print(x+y)"
        formatted_code = formatter.format_code_with_custom_style(code)
        self.assertEqual(formatted_code, expected_code)

    def test_format_code_multiple_arguments(self):
        formatter = CodeFormatter()
        code = "def multi_param_function(a,b,c,d):return a+b+c+d"
        expected_code = "def multi_param_function(a,b,c,d):return a+b+c+d"
        formatted_code = formatter.format_code_with_custom_style(code)
        self.assertEqual(formatted_code, expected_code)

    def test_format_code_multiline(self):
        formatter = CodeFormatter()
        code = "def long_function():\n    a=1\n    b=2\n    return a+b"
        expected_code = "def long_function():\n    a=1\n    b=2\n    return a+b"
        formatted_code = formatter.format_code_with_custom_style(code)
        self.assertEqual(formatted_code, expected_code)

    def test_format_code_already_correct(self):
        formatter = CodeFormatter()
        code = "def already_correct_function(a, b): return a + b"
        expected_code = "def already_correct_function(a, b): return a + b"
        formatted_code = formatter.format_code_with_custom_style(code)
        self.assertEqual(formatted_code, expected_code)

    def test_format_code_indentation(self):
        formatter = CodeFormatter()
        code = "def function():\na=1\n    b=2\n    return a+b"
        expected_code = "def function():\na=1\n    b=2\n    return a+b"
        formatted_code = formatter.format_code_with_custom_style(code)
        self.assertEqual(formatted_code, expected_code)

if __name__ == "__main__":
    unittest.main()

File: code/test_suites/cpp/FunctionBuildTest/tests/test_autopep8_format_code_with_custom_style.cpp
#include <gtest/gtest.h>
#include "../src/function_autopep8_format_code_with_custom_style.cpp"

class TestAutoPep8CustomStyle : public ::testing::Test {
protected:
    CodeFormatter formatter;
};

TEST_F(TestAutoPep8CustomStyle, TestFormatCodeBasic) {
    std::string code = "def bad_function(x,y):print(x+y)";
    std::string expected_code = "def bad_function (x, y) : print (x + y)";
    std::string formatted_code = formatter.format_code_with_custom_style(code);
    EXPECT_EQ(formatted_code, expected_code);
}

TEST_F(TestAutoPep8CustomStyle, TestFormatCodeMultipleArguments) {
    std::string code = "def multi_param_function(a,b,c,d):return a+b+c+d";
    std::string expected_code = "def multi_param_function (a, b, c, d) : return a + b + c + d";
    std::string formatted_code = formatter.format_code_with_custom_style(code);
    EXPECT_EQ(formatted_code, expected_code);
}

TEST_F(TestAutoPep8CustomStyle, TestFormatCodeMultiline) {
    std::string code = "def long_function():\n    a=1\n    b=2\n    return a+b";
    std::string expected_code = "def long_function () : a = 1 b = 2 return a + b";
    std::string formatted_code = formatter.format_code_with_custom_style(code);
    EXPECT_EQ(formatted_code, expected_code);
}

TEST_F(TestAutoPep8CustomStyle, TestFormatCodeAlreadyCorrect) {
    std::string code = "def already_correct_function(a, b): return a + b";
    std::string expected_code = "def already_correct_function (a, b) : return a + b";
    std::string formatted_code = formatter.format_code_with_custom_style(code);
    EXPECT_EQ(formatted_code, expected_code);
}

TEST_F(TestAutoPep8CustomStyle, TestFormatCodeIndentation) {
    std::string code = "def function():\na=1\n    b=2\n    return a+b";
    std::string expected_code = "def function () : a = 1 b = 2 return a + b";
    std::string formatted_code = formatter.format_code_with_custom_style(code);
    EXPECT_EQ(formatted_code, expected_code);
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}